File: leetCode/fukun/101-200/lc_160.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}

var getIntersectionNode = function (headA, headB) {
    let curr = headA;
    while (curr) {
        curr.flag = true;
        curr = curr.next;
    }
    curr = headB;
    while (curr) {
        if (curr.flag === true) return curr;
        curr = curr.next;
    }
    return null;
};

let a = new ListNode(1);
let a2 = new ListNode(2);
let a3 = new ListNode(3);
let a4 = new ListNode(4);
let b = new ListNode(9);
let b2 = new ListNode(8);

a.next = a2;
a2.next = a3;
a3.next = a4;
b.next = b2;
b2.next = a3;
console.log(getIntersectionNode(a, b));

File: leetCode/XSS/lc_160.java
import java.util.List;

public class lc_160 {
    public static void main(String[] args) {

    }
    public static class Solution {
        public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
            int lenA=getListLen(headA);int lenB=getListLen(headB);
            ListNode l;
            ListNode s;
            if(lenA-lenB>0)
            {
                l=headA;
                s=headB;
            }
            else {l=headB;s=headA;}
            int dis=Math.abs(lenA-lenB);
            int c=0;
            while (l!=null&&s!=null)
            {
                if(l==s)return l;
                if(c>=dis)
                {
                    l=l.next;s=s.next;
                }
                else
                {
                    l=l.next;
                }
                ++c;
            }
            return null;
        }
        public int getListLen(ListNode a)
        {
            int c=0;
            while(a!=null)
            {
                c++;
                a=a.next;
            }
            return c;
        }
    }
}
